completed hw28-React

maksymthehuman opened the pull request:
My contact skype: maxim7996

maksymthehuman committed:
completed hw28-React

zheleznov commented on src/components/RatingStars/RatingStars.jsx:
```diff
@@ -0,0 +1,31 @@
+
+import styles from './RatingStars.module.scss';
+
+export class RatingStars extends Component {
```
This is a simple component, better to use Functional component here instead of class component

zheleznov commented on src/components/RatingStars/RatingStars.jsx:
```diff
@@ -0,0 +1,31 @@
+    } = this.props;
+    const result = [];
+
+    for (let i = 1; i <= 5; i++) {
```
5 - this value beter to save and use as a constant
example: const MAX_STARS = 5;

zheleznov commented on src/features/homePage/Filter/Search/Search.jsx:
```diff
@@ -0,0 +1,48 @@
+    searchQuery: ''
+  }
+
+  onInputChange = (event) => {
```
you can use destructuring here from event obj

zheleznov commented on src/features/homePage/Filter/Sort/Sort.jsx:
```diff
@@ -0,0 +1,44 @@
+
+import styles from './Sort.module.scss';
+
+export class Sort extends Component {
```
to func component

zheleznov commented on src/features/homePage/HomePage.jsx:
```diff
@@ -0,0 +1,181 @@
+    sortedByAscend: true
+  }
+
+  copyArray = (oldArray) => JSON.parse(JSON.stringify(oldArray));
```
Why do you need this strange method? :)
If you need to create new copy of array you can use ES6 syntax [...]

maksymthehuman replied:
Fixed it in homework 29. In hw28 some methods depend on existence of deep copies of arrays

zheleznov commented on src/features/homePage/HomePage.jsx:
```diff
@@ -0,0 +1,181 @@
+  };
+
+  sortMovies = (type) => {
+    this.resetSortProperties(type);
```
For future need to be carefull with such cases when you change the state several time. I mean one change and lower again new setState. Like first time in this.resetSortProperties(type) and later second time. Every change state will call render () method of component and rerender it and all it's children. Basicaly not always real DOM will be updated in such cases. So this  will be covered in React optimization section

zheleznov commented on src/features/homePage/MovieInfo/MovieInfo.jsx:
```diff
@@ -0,0 +1,81 @@
+
+import styles from './MovieInfo.module.scss';
+
+export class MovieInfo extends Component {
```
to func component

zheleznov commented on src/features/homePage/MoviesList/MoviesList.jsx:
```diff
@@ -0,0 +1,43 @@
+
+  return (
+    <div className={styles.moviesContainer}>
+      {movies}
```
You can simplify this to something like this

`return (
  <div className={styles.moviesContainer}>
    {moviesList.map(({ id, title, posterUrl, stars, likes }) => (
      <div key={id} className={styles.movie}>
        <MovieCard
          id={id}
          title={title}
          posterUrl={posterUrl}
          stars={stars}
          likes={likes}
          onTitleClick={onTitleClick}
          onLikesChange={onLikesChange}
          onStarsChange={onStarsChange}
        />
      </div>
    ))}
  </div>
);`

zheleznov commented on src/features/homePage/MovieInfo/MovieInfo.jsx:
```diff
@@ -0,0 +1,81 @@
+export class MovieInfo extends Component {
+  concatenateTextArray = (textArray) => textArray.join(', ');
+
+  movieDetails = (movie) => {
```
you can destructure here in arg

maksymthehuman committed:
added fixes

- changed to functional components: RatingStars, Sort, MovieInfo;
- better use of destructuring.
